mobile/views/Thread/index.js

```diff
@@ -108,47 +111,47 @@ class Thread extends Component<Props> {
       // not sure why, but the new Date() call above breaks its inference and it thinks data.thread could be
       // undefined below
       const thread = ((data.thread: any): GetThreadType);
-      return (
-        <Wrapper>
-          <ScrollView style={{ flex: 1, width: '100%' }} testID="e2e-thread">
-            <ErrorBoundary fallbackComponent={null}>
-              <CommunityHeader thread={thread} />
-            </ErrorBoundary>
 
-            <ThreadMargin>
-              <ErrorBoundary fallbackComponent={null}>
-                <Byline navigation={navigation} author={thread.author} />
-              </ErrorBoundary>
-
-              <Text bold type="title1">
-                {thread.content.title}
-              </Text>
-
-              <Text color={props => props.theme.text.alt} type="subhead">
-                {convertTimestampToDate(createdAt)}
-              </Text>
-
-              {thread.content.body && (
-                <ThreadContent
-                  rawContentState={JSON.parse(thread.content.body)}
-                />
-              )}
-            </ThreadMargin>
-
-            <ErrorBoundary fallbackComponent={null}>
-              <ActionBar
-                content={{
-                  url: `https://spectrum.chat/thread/${thread.id}`,
-                  message: `https://spectrum.chat/thread/${thread.id}`,
-                  title: 'Look at this thread I found on Spectrum',
-                }}
-              />
-            </ErrorBoundary>
+      const ScrollComponent = thread.currentUserLastSeen
+        ? InvertedScrollView
+        : ScrollView;
 
-            <ErrorBoundary>
-              <ThreadMessages navigation={navigation} id={thread.id} />
-            </ErrorBoundary>
-          </ScrollView>
+      const ScrollChildrenData = [
+        <ErrorBoundary>
+          <ThreadMessages navigation={navigation} id={thread.id} />
+        </ErrorBoundary>,
+        <ThreadContentContainer>
+          <ThreadTitle>{thread.content.title}</ThreadTitle>
+
+          <ThreadTimestamp>{convertTimestampToDate(createdAt)}</ThreadTimestamp>
+
+          {thread.content.body && (
+            <ThreadContent rawContentState={JSON.parse(thread.content.body)} />
+          )}
+        </ThreadContentContainer>,
+        <UserListItem
+          onPressHandler={() =>
+            navigation.navigate({
+              routeName: 'User',
+              key: thread.author.user.id,
+              params: { id: thread.author.user.id },
+            })
+          }
+          user={thread.author.user}
+        />,
+      ];
+
+      const ScrollChildrenRender = thread.currentUserLastSeen
+        ? ScrollChildrenData
+        : ScrollChildrenData.reverse();
+
+      return (
+        <Wrapper>
+          <ScrollComponent testID="e2e-thread">
+            {ScrollChildrenRender.map((child, i) => (
+              <React.Fragment key={i}>{child}</React.Fragment>
```

Comment:
What's all this inversion code for?

Reply:
I will try switching back to just one scrollview and use a `scrollTo()` function to manage this